Tests/AsyncHTTPClientTests/HTTPClientTests.swift

```diff
@@ -708,6 +709,23 @@ class HTTPClientTests: XCTestCase {
             }
         }
     }
+    
+    func testProxySOCKS() throws {
+        let socksBin = try MockSocksServer(expectedURL: "/socks/test", expectedResponse: "it works!")
+        let localClient = HTTPClient(eventLoopGroupProvider: .shared(self.clientGroup),
+                                     configuration: .init(proxy: .socksServer(host: "127.0.0.1")))
+        
+        do  {
+            let response = try localClient.get(url: "http://127.0.0.1/socks/test").wait()
+            XCTAssertEqual(.ok, response.status)
+            XCTAssertEqual(ByteBuffer(string: "it works!"), response.body)
+        } catch {
+            XCTFail("\(error)")
+        }
```

Comment:
Super NIT. Johannes showed me this pattern some time ago. Personally I prefer it.

```suggestion
        var response: HTTPClient.Response?
        XCTAssertNoThrow(response = try localClient.get(url: "http://127.0.0.1/socks/test").wait())
        XCTAssertEqual(.ok, response?.status)
        XCTAssertEqual(ByteBuffer(string: "it works!"), response?.body)
```